great_expectations/agent/models.py

```diff
@@ -0,0 +1,23 @@
+from typing import Literal, Union
+
+from pydantic import BaseModel, Field
+from typing_extensions import Annotated
+
+
+class EventBase(BaseModel):
+    type: str
+
+
+class RunDataAssistantEvent(EventBase):
+    type: Literal[
+        "onboarding_data_assistant_request.received"
+    ] = "onboarding_data_assistant_request.received"
+
+
+class ShutdownEvent(EventBase):
```

Comment:
no, i left it in because the pydantic union broke if it was a single object

Reply:
and if we do plan on exposing a way to shutdown an agent, we should do so via consumer tags, so we know which agent we're shutting down

Reply:
updated this to `RunCheckpointEvent`, since that's likely to be useful in the near term: [21c804e](https://github.com/great-expectations/great_expectations/pull/7837/commits/21c804ecddac11ff83efc0079ec3fe42f3768de6)